wagtail/admin/views/pages/create.py

```diff
@@ -182,13 +188,47 @@ def get_view_draft_message_button(self):
     def get_view_live_message_button(self):
         return messages.button(self.page.url, _("View live"), new_window=False)
 
+    def set_default_privacy_setting(self):
+        # privacy setting options BaseViewRestriction.RESTRICTION_CHOICES
+        default_privacy_setting = self.page.get_default_privacy_setting(self.request)
+
+        if default_privacy_setting["type"] == BaseViewRestriction.NONE:
+            # default privacy setting is public no need to do anything
+            pass
+        elif default_privacy_setting["type"] == BaseViewRestriction.LOGIN:
+            PageViewRestriction.objects.create(
+                page=self.page, restriction_type=BaseViewRestriction.LOGIN
+            )
+        elif default_privacy_setting["type"] == BaseViewRestriction.PASSWORD:
+            PageViewRestriction.objects.create(
+                page=self.page,
+                restriction_type=BaseViewRestriction.PASSWORD,
+                password=default_privacy_setting["password"],
+            )
+        elif default_privacy_setting["type"] == BaseViewRestriction.GROUPS:
+            # Create a page view restriction for groups
+            groups_page_restriction = PageViewRestriction.objects.create(
+                page=self.page, restriction_type=BaseViewRestriction.GROUPS
+            )
+            # add groups to the page view restriction
+            groups_page_restriction.groups.set(default_privacy_setting["groups"])
+        else:
+            raise ValueError(
+                "Invalid privacy setting '" + default_privacy_setting.get("type"),
+                "Unknown"
+                + "'. Choose one of the following: 'none', 'password', 'groups', 'login'.",
+            )
```

Comment:
Suggestion: Simplify the exception message, since this should come from an enum.

```suggestion
            raise ValueError(
                f"Invalid privacy setting {default_privacy_setting.get("type")!r}"
            )
```